src/platforms/native/platform.c

```diff
@@ -485,3 +487,17 @@ static void setup_vbus_irq(void)
 
 	exti15_10_isr();
 }
+
+static void wait_for_usb_host_config_request(void)
+{
+	// Toggle idle/running LED while waiting for USB serial or
+	// DFU set configuration requests
+	while (usb_get_config() == 0)
```

Comment:
I think this should be handled within the super-loop, there must be some state variable we can check to see if we have a connection an set/reset an error code with the morse subsystem somehow

Reply:
Slightly more the point while keeping another future feature in mind - is there a way this could be done non-blocking via the SysTick handler so the firmware can do other things while the condition is true?

This would entirely block the ability for the firmware to respond to an auto-programming request which would mean rewriting it in v2.0 just to get asynchronous/non-blocking behaviour, so we'd like to explore that now instead.

Reply:
Understood, can move to SysTick handler and make non-blocking. Latest commit uses the morse subsystem but still needs to be moved to SysTick handler.

Regarding a method of checking connection state, I wanted to use an already existing API function and not intrude into how USB is configured. I have not found an API function that explicitly refers to the state of the USB connection in my exploration of the code base so far.

There are options to register callbacks for different USB events. I'm not a USB expert, but I believe the first step in an enumeration sequence is the host sending the "reset" signal. There is the option to register a callback for this using `usbd_register_reset_callback()` in `usbd.h` in libopencm3.

Similarly, a USB device will enter "suspend" mode after not receiving a start-of-frame signal for three consecutive frames. Once again, a callback can be registered using `usbd_register_suspend_callback()`. Resumption of USB comms is possible with a "resume" event. Again, a callback can be registered with `usbd_register_resume_callback()`.

I'd appreciate your feedback and I'm happy to implement a "USB connection state". Otherwise, if there is already another more suitable API function or state variable that I'm not aware of, we can simply use it instead.

Reply:
Being in suspend mode also means we are supposed to consume very little power, so i don't think ot would be appropriate here

Reply:
Yes, the spec suggests device current shouldn't exceed 50mA in this state. Without looking further right now, I assume the BMP reports to the host that it does not implement this functionality, and therefore the host does not request this mode.

Another possibility would be to monitor the start-of-frame count and assert a USB disconnection after an elapsed time with a static SOF count.

The original issue requests that an LED blinks until the BMP is enumerated. Checking `usb_get_config()` addresses this and works for the described use case of connecting the BMP with power-only USB cable. I'm happy to expand on this to present a "USB connection state" if you think it will be useful in the future.

Reply:
We actually don't signal either way to the host beyond saying we're bus powered:
```
    bmAttributes         0x80
      (Bus Powered)
```

Personally, we're fine with checking if the probe is in the configured state or not as the way this is checked as this should cover USB reset events among others and requires the USB enumeration process to not only have completed but for the host to have driven the probe into the configured state and therefore have bound drivers to the interfaces it can.